assembly/src/assembler.rs

```diff
@@ -73,7 +73,7 @@ use crate::{
 #[derive(Clone)]
 pub struct Assembler {
     /// The source manager to use for compilation and source location information
-    source_manager: Arc<dyn SourceManager>,
+    source_manager: Arc<dyn SourceManagerSync>,
```

Comment:
As discussed offline, I think we want to make the `Assembler` generic over `SourceManager`